### ISSUE TITLE
Audit for changing labels from size to complexity

### ISSUE THREAD

**Issue by SAUMILDHANKAR** [hackforla/website#2777]
### Overview
As a developer, I would like to track the size of an issue in terms of number of hours. In order to achieve that, we need to perform an audit to identify where size labels are currently being used in automations or other files before changing it to complexity.

### Details
We are planning to edit the following labels:
- ```Size: Good second issue``` to ```Complexity: Good second issue```
- ```Size: Small``` to ```Complexity: Small```
- ```Size: Medium``` to ```Complexity: Medium```
- ```Size: Large``` to ```Complexity: Large```

However, before doing that we want to document all the automations that might be dependent on these existing labels, so that no workflows are affected by this change.

### Action Items
- [x] Review all automations in [GitHub Actions Workflow Directory](https://github.com/hackforla/website/tree/gh-pages/.github/workflows) and [GitHub actions folder](https://github.com/hackforla/website/tree/gh-pages/github-actions) to figure out where size labels are being used
  - [x] Once you have the updated code for the project in VS code, you can use the find function to search the above directories for the names of the existing labels (provided under Details above)
- [x] If there are more than 2 Create a spreadsheet (otherwise just list the info in the comments below) to list all the GitHub actions where size labels are being referenced
- [x] Document any other risks that might arise due to changing of labels
- [x] Suggest a timeline for changing the labels as well as a process to revert in case existing functionalities are impacted
- [x] **For merge team**: Once this issue is done and closed, please update and edit #3109 by
  - [x] Checking off the dependency for this issue
  - [x] Editing any action items if required
  - [x] If all dependencies are checked off, please move issue #3109 to the New Issue Approval column, remove the Dependency label and add ready for milestone label.

### Resources/Instructions
[GitHub actions folder](https://github.com/hackforla/website/tree/gh-pages/github-actions)
[GitHub Actions Workflow Directory](https://github.com/hackforla/website/tree/gh-pages/.github/workflows)
[GHA architecture spreadsheet](https://docs.google.com/spreadsheets/d/12NcZQoyGYlHlMQtJE2IM8xLYpHN75agb/edit#gid=1231634015)
[Events that trigger workflows](https://docs.github.com/en/actions/reference/events-that-trigger-workflows)
[Workflow syntax for GitHub Actions](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions)

**Comment by mmogri** [comment]
Progress: just self-assigned
Blockers: TBD
Availability: 2 hrs for the rest of this week
ETA: 06/15/2022

**Comment by mmogri** [comment]
Progress: minimal (work conference and family obligations past couple weeks)
Blockers: TBD
Availability: >10 hours
ETA: 06/25/2022

**Comment by mmogri** [comment]
Progress: minimal (family obligations past couple weeks)
Blockers: TBD
Availability: >10 hours
ETA: 07/02/2022

**Comment by mmogri** [comment]
Progress: some
Blockers: TBD
Availability: >10 hours
ETA: 07/15/2022

**Comment by kathrynsilvaconway** [comment]
@mmogri 
Hi, this issue has not been updated in over 2 weeks. If you are still interested in working with HfLA, please let us know by updating your ETA and Availability. If we don't hear back within 7 days, we will unassign you from this issue and put it back in the queue. Thanks!

**Comment by kathrynsilvaconway** [comment]
This dev has left HfLA or has not responded to messages for 7 days. I'm returning this issue the the Prioritized Backlog.

**Comment by gstemmann** [comment]
Availability: 6 hours
ETA: Oct 24, 2022

**Comment by gstemmann** [comment]
Sizes are being referenced in 2 JS files (So, there is no spreadsheet required, according to the action items in the issue, and no changes have been made to the code.)

Both instances are in the [Github Actions Folder](https://github.com/hackforla/website/tree/gh-pages/github-actions).

1. [post-labels-comment.js](https://github.com/hackforla/website/blob/gh-pages/github-actions/trigger-issue/add-missing-labels-to-issues/post-labels-comment.js) --> contains many actions that utilize labels using the 'size' attribute one time in the declaration
2. [check-labels.js](https://github.com/hackforla/website/blob/gh-pages/github-actions/trigger-issue/add-missing-labels-to-issues/check-labels.js) ---> also contains processes that utilize labels using the 'size' attribute one time in the declaration

Both Files use 'size', so this will need to be considered before changing the tags to 'Complexity:'

I will also note that the entire phrases, 
- ```Size: Good second issue``` to ```Complexity: Good second issue```
- ```Size: Small``` to ```Complexity: Small```
- ```Size: Medium``` to ```Complexity: Medium```
- ```Size: Large``` to ```Complexity: Large```
were not used throughout the 2 directories listed in the issue details above. Just the use of "Size:" and labels.

The suggested timeline for changing the labels as well as a process is still unclear due to multiple automations that may be involved outside of these 2 directories.

**Comment by gstemmann** [comment]
Progress: The issue is being reviewed by the merge team to be ready to close, since this type of issue does not require a Pull Request.
Blockers: N/A
ETA: 10/30/2022

**Comment by arpitapandya** [comment]
Hey @gstemmann, Thank you for working on this issue and leaving notes.

Here are my notes to add for the next developer to work on #3109.

Currently, we have GHA that first checks for labels. If there's a missing `Size` series label, then it throws a `size: missing` label. It will post a [comment](https://github.com/hackforla/website/blob/gh-pages/github-actions/trigger-issue/add-missing-labels-to-issues/add-labels-template.md) as a reminder.

Suggestions & timeline for changes:
Consider #3109 a large issue to make all the necessary changes at once instead of breaking them into small issues and making different changes at different times. While we make changes, we must document all the necessary changes to our wiki pages & contributing guide as well(if required). This will avoid any confusion for new devs. Please, merge everything with 1 PR only.

  - [ ] Replace the labels as per the details and action items
  - [x] update necessary changes to both files in the GitHub Action folder
  - [x] Test locally
  - [x] Check the contributing guide, update if necessary
  - [ ] Check the label missing, and size series and replace that too. [resource](https://github.com/hackforla/website/wiki/How-to-read-and-interpret-labels)
  - [ ] Bonus question: Discuss why we have the `complexity: missing` label with lead developers.
  - [ ] Double check if it affects any way to pre-work checklist templates in this [folder](https://github.com/hackforla/website/tree/gh-pages/.github/ISSUE_TEMPLATE)
  - [ ] Make a plan and bring it as a demo of your research with the merge team before creating a PR
